examples/nutrients_plankton_bacteria_detritus.jl

```diff
@@ -0,0 +1,226 @@
+#####
+##### Definition of NutrientsPlanktonBacteriaDetritus
+#####
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: znode
+using Oceananigans.Fields: ZeroField, ConstantField
+using Oceananigans.Biogeochemistry: AbstractBiogeochemistry
+
+import Oceananigans.Biogeochemistry: required_biogeochemical_tracers, biogeochemical_drift_velocity
+
+"""
+    NutrientsPlanktonBacteriaDetritus(; kw...)
+
+Return a four-tracer biogeochemistry model for the interaction of nutrients, plankton
+bacteria, and detritus.
+"""
+Base.@kwdef struct NutrientsPlanktonBacteriaDetritus{FT} <: AbstractBiogeochemistry
+    maximum_plankton_growth_rate :: FT = 1/day
+    maximum_bacteria_growth_rate :: FT = 0.5/day
+    bacteria_yield :: FT               = 0.2
+    quadratic_mortality_rate :: FT     = 1/day # m³/mmol/day
+    nutrient_half_saturation :: FT     = 0.1   # mmol m⁻³
+    detritus_half_saturation :: FT     = 0.1   # mmol m⁻³
+    PAR_half_saturation :: FT          = 10.0  # W m⁻²
+    PAR_attenuation_scale :: FT        = 25.0  # m
+    detritus_sinking_speed :: FT       = 10/day # m s⁻¹
+end
+
+const NPBD = NutrientsPlanktonBacteriaDetritus
+
+@inline required_biogeochemical_tracers(::NPBD) = (:N, :P, :B, :D)
+
+@inline function biogeochemical_drift_velocity(bgc::NPBD, ::Val{:D})
+    u = ZeroField()
+    v = ZeroField()
+    w = ConstantField(- bgc.detritus_sinking_speed)
+    return (; u, v, w)
+end
```

Comment:
Is this actually used somewhere? Unclear how this instruction is applied to the model or used in the simulation.

Reply:
The function `biogeochemical_drift_velocity` returns the "additional" velocity field that's applied to the tracer `D`. So here we are saying `D` is advected not only by `model.velocities` (note we could _also_ have a "background velocity" if we use the nonhydrostatic model) but also by `(; u, v, w)`.

The function is imported on this line:

https://github.com/glwagner/PeachySea.jl/blob/5809a51fc547db49543bbb1d1866d68dc12f81c4/examples/nutrients_plankton_bacteria_detritus.jl#L11

which makes it available for "extension" via dispatch.

The function is called deep within Oceananigans, here:

https://github.com/CliMA/Oceananigans.jl/blob/bcc34f07b3f949ea6fb34c7814f4b856d24924c2/src/Biogeochemistry.jl#L67-L75

The default is `nothing`:

https://github.com/CliMA/Oceananigans.jl/blob/bcc34f07b3f949ea6fb34c7814f4b856d24924c2/src/Biogeochemistry.jl#L29

Finally, note that this will change slightly (to improve the numerics) with https://github.com/CliMA/Oceananigans.jl/pull/3027. Specifically we need to sum the advecting velocities prior to calculating the discrete, upwinded advected flux. Otherwise upwinding may not be correctly applied.

Reply:
Advice or PRs that improve the design or documentation are definitely welcome!

Reply:
This is great! If I understand correctly this allows us to easily add w_sinking to the background w in, for example, a 3D ocean circulation. I'll note now, in case it's important for programming now, that moving forward (for the non-"simple" bgc model), it'd be great to have an array of Ds that each have their own w_sinking. In the simplest case, having two pools of D, one with w_sinking > 0 and one with w_sinking = 0, would represent "POM" (particulate organic matter) and "DOM" (dissolved organic matter).

Reply:
> 

Thanks for the clarification - I was looking for `biogeochemical_drift_velocity ` to be declared somewhere like `HydrostaticFreeSurfaceModel`